fiboa_cli/convert_utils.py

```diff
@@ -42,13 +42,18 @@ def convert(
         geoparquet1 = False,
         original_geometries = False,
         index_as_id = False,
+        mapping_file = None,
+        open_options = None,
         **kwargs):
     """
     Converts a field boundary datasets to fiboa.
     """
     if bbox is not None and len(bbox) != 4:
         raise ValueError("If provided, the bounding box must consist of 4 numbers")
 
+    if open_options:
+        kwargs.update(open_options)
+
```

Comment:
Shouldn't that be possible in the converter code by updating kwargs there? So ideally we don't need another parameter...

Reply:
Yes, but sadly my keyword-argument for the gdf-open function overlaps with the arguments of convert. I'm trying to pass `columns=["col1", "col2"]` so only a selection is loaded into the not-lazy data frame. That way I can keep the memory usage down (it now becomes 'only' ~65GB).

Reply:
This pandas dataframe in-memory transformation has it's limits. I've worked with 'lazy' dataframes, I think it was with polars instead of pandas. No straightforward solution to this problem I guess https://pandas.pydata.org/docs/user_guide/scale.html ...

Reply:
Yeah, my comment was about the additional parameters, this would get less bad with the class-based approach.

The memory issue is indeed yet another topic that will occur more and more with larger datasets.

Reply:
Just wondering, but I think we could just rename our parameter to column_map or so. We never pass it by name, so a rename should only really have an effect in the convert_utils, not in the converters. That would make the conflict go away, which could generally be useful @ivorbosloper

Reply:
> I think we could just rename our parameter to column_map or so. We never pass it by name.

Ah, hadn't seen this. Would keep it a bit simpler I suppose. Will rewrite it.